### PR TITLE
add validation for channel and upstream

### PR DESCRIPTION
**Pull request by leena-jawale**
Signed-off-by: Leena Jawale <ljawale@redhat.com>

Updates: 
- Add validation for channel and upstream
- Add unit test %
- remove makefile.prow
- updated dockerfile to use golang 1.16
- added check to add pre post hook only if pre post specified in curator CR

### CODE REVIEW

File: cmd/curator/curator.go [hunk 1]
```diff
@@ -225,17 +225,34 @@ func curatorRun(config *rest.Config, client *clientv1.Client, clusterName string
 				jobChoice,
 				v1.ConditionTrue,
 				err.Error()))
+			klog.Error(err.Error())
+			panic(err)
+		}
+	}
+
+	if jobChoice == "prehook-ansiblejob" {
```
Comment (jnpacker) [inline cmd/curator/curator.go:233]: You don't want to just do an OR for posthook-ansiblejob or prehook-ansiblejob so we don't get a code smell?

File: pkg/jobs/hive/hive.go [hunk 2]
```diff
@@ -426,5 +430,35 @@ func validateUpgradeVersion(client clientv1.Client, clusterName string, curator
...
+		}
+	}
+	if channel != "" && !isValidChannel {
+		return errors.New("Provided channel is not valid")
+	}
+
+	isValidUpstream := false
+	if channel != "" {
+		for _, versionRelease := range managedClusterInfo.Status.DistributionInfo.OCP.VersionAvailableUpdates {
+			if versionRelease.Version == desiredUpdate && versionRelease.URL == upstream {
```
Comment (jnpacker) [inline pkg/jobs/hive/hive.go:453]: Any reason to not put the versionRelease.URL == upstream between line 439 and 440 above to save some looping?

File: cmd/curator/curator.go (hunk 1 above)
Comment (leena-jawale) [reply cmd/curator/curator.go:233]: i can combine these 2 if for pre and post

**Pull request merged**